enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,335 @@
+	// to contain at least one of these strings.
+	// +optional
+	Models []string `json:"models"`
+
+	// Vendors is a list of device vendors. If not empty, the device's model as outputted by lsblk needs
+	// to contain at least one of these strings.
+	// +optional
+	Vendors []string `json:"vendors"`
+}
+
+type LocalVolumeGroupPhase string
+
+const (
+	ActivePhase LocalVolumeGroupPhase = "Active"
+	FailedPhase LocalVolumeGroupPhase = "Failed"
+)
+
+type LocalVolumeGroupStatus struct {
+	// Phase describes the state of the LocalVolumeGroup
+	Phase LocalVolumeGroupPhase `json:"phase,omitempty"`
+
+	// Conditions is a list of conditions and their status.
+	Conditions []operatorv1.OperatorCondition `json:"conditions,omitempty"`
+
+	// A human-readable message indicating details about why the LocalVolumeGroup is in this state.
+	// +optional
+	Message string `json:"message,omitempty"`
+
+	// Reason is a brief CamelCase string that describes any failure and is meant
+	// for machine parsing and tidy display in the CLI.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+
+	// TotalProvisionedDeviceCount is the count of the total devices over which the PVs has been provisioned
+	TotalProvisionedDeviceCount *int32 `json:"totalProvisionedDeviceCount,omitempty"`
+
+	// observedGeneration is the last generation change the operator has dealt with
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+}
+```
+### Workflow of LocalPV creation via LocalVolumeGroup CR
+- Once the `LocalVolumeGroup` is created, the controller will:
+  - configure the local-static-provisioner to make a new StorageClass based on certain directories on the selected nodes.
+  - assign diskmaker daemons to the selected nodes.
+- The diskmaker daemon will find devices that match the disovery policy and symlink them into the directory that the local-static-provisioner is watching.
+- Possible `status.phase` values for `LocalVolumeGroup`
+  - `Active`
+  - `Failed`
+
+#### Note: There is a chance of race condition and duplicate creation of PVs 
+- If two `LocalVolumeGroup` CR targets same nodes with overlapping inclusion filter.
+- If `LocalVolumeGroup` CR and `LocalVolume` CR targets the same node with overlapping devices.
+  
+Example of `LocalVolumeGroup` CR:
+
+```yaml
+apiVersion: local.storage.openshift.io/v1alpha1
+kind: LocalVolumeGroup
+metadata:
+  name: example-autodetect
+spec:
+  nodeSelector:
+    nodeSelectorTerms:
+      - matchExpressions:
+          - key: kubernetes.io/hostname
+            operator: In
+            values:
+              - worker-0
+              - worker-1
+  storageClassName: example-storageclass
+  volumeMode: Block
+  minDeviceCount: 5
+  maxDeviceCount: 10
+  deviceInclusionSpec:
+    deviceTypes:
+      - Disk
+    deviceMechanicalProperty: 
+      - Rotational
+      - NonRotational
+    minSize: 10G
+    maxSize: 100G
+status:
+  phase: Active
+  totalProvisionedDeviceCount: 4
```

Comment:
Is this a real example? `minDeviceCount` was 5, why 4 is a valid number? Note that it may be a valid value when the fifth PV is being created, however, some explanation below would be appropriate.

Reply:
okay, I changed to 5. I didn't want to put an exceptional example.